test/test_Flux.jl

```diff
@@ -304,6 +304,63 @@ function test_unsupported_activation()
     return
 end
 
+function test_vector_nonlinear_oracle_errors()
+    chain = Flux.Chain(Flux.Dense(3 => 16, Flux.sigmoid), Flux.Dense(16 => 2))
+    @test_throws(
+        ErrorException(
+            "cannot specify `gray_box = true` if `vector_nonlinear_oracle = true`",
+        ),
+        MathOptAI.build_predictor(
+            chain;
+            gray_box = true,
+            vector_nonlinear_oracle = true,
+        ),
+    )
+    @test_throws(
+        ErrorException(
+            "cannot specify the `config` kwarg if `vector_nonlinear_oracle = true`",
+        ),
+        MathOptAI.build_predictor(
+            chain;
+            config = Dict(Flux.relu => MathOptAI.ReLUBigM(100.0)),
+            vector_nonlinear_oracle = true,
+        ),
+    )
+    return
+end
+
+function test_vector_nonlinear_oracle_sigmoid()
+    chain = Flux.Chain(Flux.Dense(3 => 16, Flux.sigmoid), Flux.Dense(16 => 2))
+    model = Model(Ipopt.Optimizer)
+    set_silent(model)
+    @variable(model, x[i = 1:3] == i)
+    y, formulation =
+        MathOptAI.add_predictor(model, chain, x; vector_nonlinear_oracle = true)
```

Comment:
@Robbybp seems nice?

Reply:
Yes. I will update my internal examples to use this API and will approve when I've done so.